### PR TITLE
[Zhang Dongjun] Update for quiz Ui and relevant command

### PR DESCRIPTION
**Pull request by Auxinnn**
Can be merged after reviewed. Finish the basic part of quiz Ui.

### CODE REVIEW

**Comment by shuiyao-sg** [comment]
Try not to send new PRs when the last PR you created has not been merged yet.

**Review by Parcly-Taxel — CHANGES_REQUESTED** [review]
Quiz mode doesn't work. A `NullPointerException` is raised:

```
INFO: ----------------[USER COMMAND][addq q/Q a/A d/D s/S]
INFO: ----------------[USER COMMAND][quiz n/1 d/D s/S]
INFO: Result: You have successfully entered quiz mode!
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException...
```

Making all the changes listed below would resolve this exception, but there is still no _visible_ appearance of any panel showing the question.

File: src/main/resources/view/QuizQuetionListCard.fxml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Parcly-Taxel) [inline src/main/resources/view/QuizQuetionListCard.fxml]: The file name `QuizQuetionListCard.fxml` is wrongly spelt and should be spelt `QuizQuestionListCard.fxml`.

File: src/main/java/seedu/address/ui/QuizQuestionListCard.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.question.Question;
+
+/**
+ * An UI component that displays information of a {@code Question} in quiz.
+ */
+public class QuizQuestionListCard extends UiPart<Region> {
+    private static final String FXML = "QuizQuestionListCard";
```
Comment (Parcly-Taxel) [inline src/main/java/seedu/address/ui/QuizQuestionListCard.java]: String constant should be `QuizQuestionListCard.fxml`, with the file extension.

File: src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java [hunk 3]
```diff
@@ -14,14 +18,17 @@
     /**
      * Used for initial separation of question index and answer.
      */
-    private static final Pattern BASIC_INPUT_FORMAT = Pattern.compile("(?<index>\\s+)(?<answer>)");
+    private static final Pattern BASIC_INPUT_FORMAT = Pattern.compile("(?<index>\\S+)(?<answer>.*)");
```
Comment (Parcly-Taxel) [inline src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java]: The regex should be `(?<index>\\d+):(?<answer>.+)`. This enforces that the index is numeric and that the answer is non-empty, _and_ separates the index from the answer.

File: src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java [hunk 4]
```diff
@@ -14,14 +18,17 @@
...
 
     /**
      * Parses the given {@code String} of user input in the context of the quiz answer
      * and returns an QuizCheckAnswer object for execution.
      */
-    public QuizCheckAnswer parse(String userInput) {
+    public QuizCheckAnswer parse(String userInput) throws ParseException {
         final Matcher matcher = BASIC_INPUT_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
```
Comment (Parcly-Taxel) [inline src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java]: Don't just use a generic command and help. Write an error message and help specifically for the answer.

Comment (Auxinnn) [reply src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java]: Thank you very much for your help! I will fix all problems raised above.

**Review by Parcly-Taxel — APPROVED** [review]

**Pull request merged**